jbmc/src/java_bytecode/java_object_factory.cpp

```diff
@@ -1207,6 +1207,29 @@ void java_object_factoryt::array_loop_init_code(
     plus_exprt(array_init_symexpr, counter_expr, array_init_symexpr.type()),
     array_init_symexpr.type().subtype());
 
+  bool new_item_is_primitive = arraycellref.type().id() != ID_pointer;
+
+  // Use a temporary to initialise a new, or update an existing, non-primitive.
+  // This makes it clearer that in a sequence like
+  // `new_item->x = y; new_item->z = w;` that all the `new_item` references must
+  // alias, cf. the harder-to-analyse
+  // `some_expr[idx]->x = y; some_expr[idx]->z = w;`
+  exprt init_expr;
+  if(new_item_is_primitive)
+  {
+    init_expr = arraycellref;
+  }
+  else
+  {
+    init_expr = allocate_objects.allocate_automatic_local_object(
+      arraycellref.type(), "new_array_item");
```

Comment:
Please use the same name as in the comment above (either this to `new_item`, or the comment to `new_array_item`